Lib/test/test_py2xwarn.py

```diff
@@ -0,0 +1,25 @@
+import unittest
+import sys
+from test.support.warnings_helper import check_py2x_warnings
+import warnings
+
+if not sys.py2x_warning:
+    raise unittest.SkipTest('%s must be run with the -2 flag' % __name__)
+
+class TestPy2xWarnings(unittest.TestCase):
+
+    def assertWarning(self, _, warning, expected_message):
+        self.assertEqual(str(warning.message), expected_message)
+
+    def assertNoWarning(self, _, recorder):
+        self.assertEqual(len(recorder.warnings), 0)
+
+    def test_unicode(self):
+        expected = (
+                   "the unicode type is not supported in 3.x"
+                   "use native strings for compatibility or "
+                   "a 'u' or 'b' prefix if a native string is not required"
+                   )
+        with check_py2x_warnings((expected, SyntaxWarning)):
```

Comment:
I *think* it's correct to say that in Py3 all strings are unicode, so simply stripping the `u` prefix is always correct?

Reply:
This also made me wonder if we need to port the `unicode` function over? It would, I think, be mostly (entirely?) a no-op.

Reply:
> This also made me wonder if we need to port the `unicode` function over? It would, I think, be mostly (entirely?) a no-op.

Actually unicode is ported already in 3.x but the semantics are not supported, its solely for compatibility.

Reply:
> I _think_ it's correct to say that in Py3 all strings are unicode, so simply stripping the `u` prefix is always correct?

Technically it is so, but I think the type is now called 'str'

Reply:
> Actually unicode is ported already in 3.x but the semantics are not supported, its solely for compatibility.

So it's a no-op?

Reply:
Yes, its no-op

Reply:
I don't understand this one -- what does "differed the warning" mean?

Reply:
Removed the unicode-specific warning